src/Bravo3/ImageManager/Services/ImageManager.php

```diff
@@ -62,13 +63,21 @@ class ImageManager
 
     /**
      * If true, we won't trust cache tags saying an image does not exist and will always check with the filesystem
-     * before throwing a NotExistsException
+     * before throwing a NotExistsException.
      *
      * @var bool
      */
     protected $validate_tags;
 
     /**
+     * An associated array with unique Image key as the key and ImageMetadata objects
+     * as values.
+     *
+     * @var array
+     */
+    protected static $metadata_cache;
```

Comment:
Should not be static, this will create corruption between instances of the service